### ISSUE TITLE
Can't see messages I've sent, only messages received 

### ISSUE THREAD

**Issue by ghost** [glaurent/MessagesHistoryBrowser#9]
I've used this about 3 years ago and everything worked really well. I've come back to create an updated pdf however, I can only see what my contacts have sent me, not what I've sent them. I have gone through my message archives on both my Mac and iPhone and everything seems to be there, it's just not picking up on it. Is there a setting I have to fiddle around with?

**Comment by glaurent** [comment]
You're the 2nd user to report this bug, I haven't been able to reproduce it. Would you be able to run test builds with debug traces in them to help me pinpoint the issue ?

**Comment by ghost** [comment]
Happy to help, just unsure what you need me to do exactly, don’t have any experience, but I’m sure I can follow instructions 😊

**Comment by glaurent** [comment]
Do you have Xcode installed and can you build from source ?

**Comment by glaurent** [comment]
@raraararar any news ?

**Comment by ghost** [comment]
Hey there, I had actually recently deleted Xcode cos I needed more space, I tried to reinstall it and it was unsuccessful
I have online exams soon, just need my laptop to survive that so I’ll try again in June

**Comment by glaurent** [comment]
Ok, thanks 😊. Good luck on your exams 👍 .

**Comment by plbecker** [comment]
Hi Guillaume,
I have the same issue and have Xcode installed / could run some tests

**Comment by glaurent** [comment]
Hi @plbecker , great ☺️. Are you able to build and run the app from source ? If so I'll point you to where you can put some breakpoints or traces to help me understand this issue.

**Comment by plbecker** [comment]
I get an error message when trying to run the cloned repository: "Apple Development: ambiguous matches" in my keychain (which I don't get for any of my other Xcode projects). I'll try to fix this issue and will get back to you!

**Comment by plbecker** [comment]
Hi so the issue seems to be that the messages that were sent by me are misplaces / not in the same chat timeline. By logging them, I did actually end up seeing what I sent myself, but not in the same chat. I'll try to find a workaround. For a slightly different purpose, I don't need the UI from your app, just the fetching / iMessage reading part (otherwise I'd just do a PR)

**Comment by glaurent** [comment]
@plbecker wow, thanks a lot, I didn't expect you'd investigate the issue 👍 . If you're looking at just fetching the messages, take a look at the scripts I've used to extract the messages from the chat.db. They are linked from the app's About panel :

Initial Messages DB info from 
Matthieu Riegler https://github.com/kyro38/MiscStuff/blob/master/OSXStuff/iMessageBackup.sh

and Noah Slater : https://gist.github.com/nslater/b3cbc894ad2c2516dd02

**Comment by plbecker** [comment]
Perfect, thank you very much! I'll look into those codebases. If I find the issue that prevents my messages from being shown too, I'll let you know!

**Comment by glaurent** [comment]
Can you post some logs that could point me to the issue, without showing any personal info from your messages db ?

**Comment by plbecker** [comment]
Yes of course, I'll send you a private email with this information

**Comment by glaurent** [comment]
Thanks a lot :) You can send it to the email address in my github profile.

**Comment by glaurent** [comment]
Hi @plbecker, any updates ?

**Comment by plbecker** [comment]
Hey, I sent you an email! Sorry for the delay

**Comment by glaurent** [comment]
Got it, thanks a lot. I'll take a look at my code and see if I can work something from there.

**Comment by ulope** [comment]
I'm having the same issue. 
In the specific conversation I'm looking at everything older than about a year and a half I only see messages from the other person.

**Comment by glaurent** [comment]
@ulope - I never got enough info from @plbecker unfortunately, so this issue is still pending. Would you be able to run debug builds of the app ? Or to run it directly from Xcode ?

**Comment by ulope** [comment]
@glaurent Yes I can do that. I might need some pointers what to look for though. Swift isn't my home turf ;)

**Comment by glaurent** [comment]
@ulope great, let me push a few build fixes quickly (I haven't had time to maintain this code in a while), and get back to you.

**Comment by glaurent** [comment]
@ulope if you can add some traces (using print() or NSLog()) in ChatsDatabase:collectMessagesForChat, specifically around l264

`            for messageData in query {
                let messageContent = messageData[textColumn] ?? ""
                var dateInt = messageData[dateColumn]
...
`

so I can see if the messages from yourself are not returned because they're not in the SQL query or because of some other issue.

**Comment by ulope** [comment]
@glaurent I'm having a bit of trouble getting it to compile.
It's missing SQLite.

I did run `pod install` and then manually changed the target of the `Sparkle.framework` to point to the one in the `Pods` directory.
Unfortunately for SQLite there is no `.framework` to be found in `Pods`.

Do you have an idea what I might be doing wrong?

**Comment by glaurent** [comment]
After `pod install`, you have to open the `.xcworkspace` file that has been created instead of the `.xcodeproj`. Revert all the changes you've made in the xcodeproj, once pods are setup, it's not meant to be opened on its own.

(sorry, you're getting a crash course in macOS development because of this 😕. Thanks for taking the time to work on it)

**Comment by ulope** [comment]
Ok that helped, thanks.

However I neither see `print` / `NSLog` output nor do breakpoints seem to work in any of the methods in `ChatsDatabase` except `init`.

I'm sure I must be missing something obvious...

> (sorry, you're getting a crash course in macOS development because of this 😕. Thanks for taking the time to work on it)

Nah, don't worry about it. It's good for the ego to feel like a dummy again every once in a while :D

**Comment by glaurent** [comment]
Have you tried re-importing your messages DB ? The other methods won't be called otherwise.
